### DISCUSSION TITLE
Unable to build project on an M1 iMac

### DISCUSSION THREAD

**Discussion by CrystalKnightCodes** [pointfreeco/swift-composable-architecture#571] [Q&A]
I am building an app that uses TCA, and I recently purchased the new iMac with the M1.  I can successfully build and run the project on my MacBook Pro, but when I try to run the same code, same branch, same device (all updated to the same versions), I get a build error on the iMac.
macOS Big Sur 11.3.1
Xcode 12.5
iOS: 14.5.1 iPhone 11 Pro Max

[Code](https://gist.github.com/CrystalKnightCodes/2c86d57c8f3f504e392b844c50392ec3)

Error message on line 23: 
Cannot convert value of type 'Store<IdentifiedArray<UUID, HotlistSelection>, HotlistSelector.ListActionData>' (aka 'Store<IdentifiedArray<UUID, HotlistSelection>, (id: UUID, action: HotlistManagementAction.List)>') to expected argument type 'Store<IdentifiedArray<UUID, HotlistSelection>, (UUID, HotlistManagementAction.List)>'

Since the code runs fine on the MacBook Pro, I suspect this has to do with building the library on ARM.  Does anyone have a workaround?

**Comment by stephencelis** [comment]
A few others have had issues with this in the past depending on project setup, _e.g._: https://github.com/pointfreeco/swift-composable-architecture/issues/344

Hopefully you get to the bottom of it!
